core/src/main/java/com/linecorp/armeria/client/endpoint/CompositeEndpointGroup.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.endpoint;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.Closer;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.util.AbstractListenable;
+
+/**
+ * An {@link EndpointGroup} that merges the result of any number of other {@link EndpointGroup}s.
+ */
+public final class CompositeEndpointGroup extends AbstractListenable<List<Endpoint>> implements EndpointGroup {
```

Comment:
How about making this package-private and adding `EndpointGroup.of(EndpointGroup...)` and `of(Iterable<EndpointGroup>)`?

Reply:
Thanks - will make the change but putting this PR on hold until #1940

Reply:
Merged #1940 :-)